Sources/MapboxNavigation/BuildingHighlighting.swift

```diff
@@ -49,8 +49,9 @@ extension BuildingHighlighting {
         
         // In case if not all buildings were found and user is within allowed destination
         // threshold - attempt to highlight buildings based on destination coordinate.
+        let destination = progress.currentLeg.destination
         if !buildingWasFound, passedApproachingDestinationThreshold,
-           let currentLegWaypoint = progress.currentLeg.destination?.targetCoordinate {
+           let currentLegWaypoint = destination?.targetCoordinate ?? destination?.coordinate {
```

Comment:
The `targetCoordinate` may be lost after the rerouting event

Reply:
I don't think that usage of final destination coordinate is a reliable source. In most of the cases it'll contain coordinate that is snapped to the road network, and building identifier querying will fail. Would it be possible to copy `targetCoordinate` right before rerouting event? If not, we should discuss this with Navigation Native folks on how it can be improved.

Reply:
I see. I'll ask Navigation Native about that and remove this one. The query failed with the destination coordinate.